File: Checkpoint-Manager/src/aber/dcs/cs22510/clg11/model/Node.java
package aber.dcs.cs22510.clg11.model;

/**
 * Defines the data model for a course node within an event. 
 * Allows the setting and retrieval of data about a particular course node.
 * 
 * @author Connor Luke Goddard (clg11)
 * Copyright: Aberystwyth University, Aberystwyth.
 */
public class Node {
    
    /** Type of the node. (CP, MC, JN) */
    private String type;
    
    /** The unique node number. */
    private int number;
    
    /**
     * Default constructor for a Node.
     */
    public Node() {
        
    }
    
    /**
     * Constructor for a Node that allows their characteristics to be set upon 
     * instantiation.
     * 
     * @param cpNumber The new node number to be set.
     * @param cpType The node type of the new node 
     */
    public Node(int cpNumber, String cpType) {
        
        this.number = cpNumber;
        this.type = cpType;
        
    }

    /**
     * Returns the node type of the current node.
     * @return The type of the current node.
     */
    public String getType() {
        return type;
    }

    /**
     * Set the current node type.
     * @param type The new node type to be set.
     */
    public void setType(String type) {
        this.type = type;
    }

    /**
     * Returns the ID number of the node.
     * @return The number of the current node.
     */
    public int getNumber() {
        return number;
    }

    /**
     * Sets the ID number of the current node.
     * @param number the number to set
     */
    public void setNumber(int number) {
        this.number = number;
    }
    
}

File: Event-Creator/Node.cpp
/**
 * File: Node.cpp
 * Description: Provides the data model for a particular course node.
 * Author: Connor Luke Goddard (clg11)
 * Date: March 2013
 * Copyright: Aberystwyth University, Aberystwyth
 */

#include <iostream>
#include "Node.h"

using namespace std;

/**
 * Constructor that allows the characteristics of a course node to be specified.
 * Constant variable 'nodeNo' is set it's value here.
 * @param newNodeNo The unique identifier of a particular node. (constant)
 * @param newNodeType The course node type to be set.
 */
Node::Node(const int newNodeNo, string newNodeType) : nodeNo(newNodeNo){
   
    setNodeType(newNodeType);
    
}

/**
 * Destructor to be used once object is removed.
 */
Node::~Node() {
}


/**
 * NOTE: setNodeNo() cannot be used due to 'nodeNo'
 * being a CONSTANT value. It therefore cannot be changed
 * once created. Making the variable MUTABLE however would allow ]
 * it to be changed.
 */

/**
 * Updates the type value of the node.
 * @param nodeType The new node type value.
 */
void Node::setNodeType(string nodeType) {
    this->nodeType = nodeType;
}

/**
 * Fetches the node type of the current node.
 * @return The type of the current node.
 */
string Node::getNodeType(void) const {
    return nodeType;
}

/**
 * Fetches the unique number of the node.
 * @return The number representing the node.
 */
const int Node::getNodeNo(void) const {
    return nodeNo;
}